File: Java/Collections/SetDemo.java
package Java.Collections;

import java.util.HashSet;
import java.util.Set;
import java.util.TreeSet;

public class SetDemo {
    public static void main(String[] args){

        Set<Integer> valuesTree = new TreeSet<>();
        Set<Integer> valuesHash = new HashSet<>();

        valuesHash.add(24);
        valuesTree.add(24);
        valuesHash.add(14);
        valuesTree.add(14);
        valuesHash.add(54);
        valuesTree.add(54);
        valuesHash.add(22);
        valuesTree.add(22);

        // Add duplicate
        valuesHash.add(22);
        valuesTree.add(22);
        // Set cannot have duplicate values

        // HASH set is random order
        for(Integer i : valuesHash){
            System.out.println(i);
        }

        // Treeset is increasing order
        for(Integer i : valuesTree){
            System.out.println(i);
        }

        // Check if contains an element
        valuesHash.contains(123);
    }
}

File: Kotlin/Collections/SetDemo.kt
package Kotlin.Collections

import java.util.*
import kotlin.collections.HashSet

fun main(args :Array<String>){
   /* A set is an unordered collection that does not support duplicates.
    For creating sets, there are functions setOf() and mutableSetOf().
    A read-only view of a mutable set can be obtained by casting it to Set.*/

    val set : Set<String> = setOf<String>("asd","Asdas","asdasda")
    // set.add not available

    val mutableSet : MutableSet<String> = mutableSetOf("asdasd","asdasd","adasdasd")
    val isElementAdded = mutableSet.add("asdasd")
    // isElementAdded false if element is already there

    // Creating an empty set
    val emptySet = emptySet<String>()
    val emptySet1 = setOf<String>()
    val emptySet2 = mutableSetOf<String>()


    // Hashset mutable by default
    val hasSet = HashSet<String>()
    val hasSet1 = hashSetOf<String>()
    hasSet.add("123")

    // Treeset mutable by default
    val treeSet = TreeSet<String>()
    treeSet.add("123123")

    // Iterate a set
    set.forEach {
        println(it)
    }
    mutableSet.forEach {
        println(it)
    }
    hasSet.forEach {
        println(it)
    }

    // check if set contains an element
    set.contains("asdas")
    mutableSet.contains("Asd")
    hasSet.contains("Asd")
}